src/store/index.ts

```diff
@@ -0,0 +1,91 @@
+import Vue from 'vue';
+import Vuex, { Store } from 'vuex';
+import { createDirectStore } from 'direct-vuex';
+
+import api from '@/api';
+
+Vue.use(Vuex);
+
+interface WorkspaceSpec {
+  name: string;
+  nodeTables?: string[];
+  edgeTables?: string[];
+  graphs?: string[];
+}
+
+interface State {
+  workspaces: string[];
+  currentWorkspace: WorkspaceSpec | null;
+}
```

Comment:
Maybe one or both of these would be best defined in a global types.ts?

Reply:
Good point, I think it makes sense to move `WorkspaceSpec` there, but I think `State` is fine where it is, as it's inherent to the store.